### ISSUE TITLE
Instances returned by get due to django_get_or_create don't apply kwargs

### ISSUE THREAD

**Issue by carn1x** [FactoryBoy/factory_boy#483]
Model
```
class Currency(models.Model):
    iso = models.CharField(max_length=4, unique=True)
    fee = models.DecimalField(max_digits=8, decimal_places=2)
```

Factory
```
class CurrencyFactory(DjangoModelFactory):

    iso = factory.LazyAttribute(lambda x: fake.currency_code())

    class Meta:
        model = Currency
        django_get_or_create = ('iso',)
```

Test
```
>>> C(iso='HKD').fee
Decimal('10.00')
>>> C(iso='HKD', fee=5).fee
Decimal('10.00')
>>> C(iso='ABC', fee=5).fee
Decimal('5.00')
>>> C(iso='ABC', fee=6).fee
Decimal('5.00')
```

Despite the overriding value provided in the factory constructor, the overriding values are ignored. Is this by design and the docs should ideally clarify this or is this an oversight? It's particularly annoying to debug this issue in a large test suite that may run in a random order and so sometimes collide.

**Comment by carn1x** [comment]
ahh nice info, thanks very much!

**Comment by madEng84** [comment]
Same problem here

@rbarrois your solution is ok, but i think we can transform this issue in a feature request to add `django_update_or_create` in the Factory.Meta or to modify `django_get_or_create` to modify the instance taken from DB

I think that `django_get_or_create` should works as an update_or_create because when i istantiate a factory with fee=5, in my expectation the fee should be 5, am I wrong?
What do you think about?

Thank you in advance ;D

**Comment by francoisfreitag** [comment]
For reference, https://github.com/FactoryBoy/factory_boy/pull/558 suggested introducing `django_update_or_create`. It caused concerns that `update_or_create` makes tracking the current value for a row in the DB harder. The DB value could be updated for each factory call, updating the DB row as a side effect, making it harder to predict.

Consider the following example:
```pycon
>>> CurrencyFactory(iso='HKD')
Fee(iso='HKD', fee=Decimal('10.00'))
>>> # Using django_update_or_create, update the database record for fee.
>>> CurrencyFactory(iso='HKD', fee=5)
Fee(iso='HKD', fee=Decimal('5.00'))
>>> CurrencyFactory(iso='HKD')
Fee(iso='HKD', fee=Decimal('5.00'))
```

I agree with Federico, this behavior is very likely to cause confusion.

**Comment by madEng84** [comment]
@francoisfreitag thanks for the reference

In the example you wrote I don't see any confusion, because you have defined the `get_or_create` attribute in your Factory and you have changed that object in your test case.

The real confusion, according to me, is only this:
```
>>> CurrencyFactory(iso='HKD')
Fee(iso='HKD', fee=Decimal('10.00'))
>>> CurrencyFactory(iso='HKD', fee=decimal.Decimal('5'))
Fee(iso='HKD', fee=Decimal('10.00')) #  :O
```

The problem you're speaking about becomes confusing if you share the changed database within different tests and you instantiate your Factory in the setUp method or in a fixture, but i think the single test should tell the entire history so I always try to avoid using fixtures (in order to ensure a better readability)

> The DB value could be updated for each factory call

Yes, because the Factory, if called, should hit the DB, otherwise you would call the Factory's build() method, and if you want that object from the DB you can always use the manager methods

Basically, according to me, get_or_create method should be substituted internally by update_or_create if the object exists, and the build() method should change the obj from DB if existing without saving at DB

If I find a little bit of time, i will make a PR, than we can speak about it referencing our discussion to the tests I will write

But i want to know what do @rbarrois and @jeffwidman  think about this discussion, too.

Regards

**Comment by mattduck** [comment]
Hi! FWIW I lost some time today debugging this exact situation. I think the reasoning for not implementing the feature as `update_or_create` is fair, but it violates expectation to explicitly pass in a field value and receive back an object with a different value.

+1 for at least updating the docs to make it clear that kwargs are not applied when `get_or_create` is used. I'll look into submitting a patch for the docs this weekend (but can't promise I'll get to it).
